src/guake/prefs.py

```diff
@@ -726,19 +729,57 @@ def reload_erase_combos(self, btn=None):
             if ERASE_BINDINGS.get(i[0]) == binding:
                 combo.set_active_iter(i.iter)
 
-    def load_configs(self):
-        """Load configurations for all widgets in General, Scrolling
-        and Appearance tabs from gconf.
-        """
-        # default_shell
-
+    def _load_hooks_settings(self):
+        """load hooks settings"""
+        log.debug("executing _load_hooks_settings")
+        hook_show_widget = self.get_widget("hook_show")
+        hook_show_setting = self.client.get_string(KEY("/hooks/show"))
+        if hook_show_widget is not None:
+            if hook_show_setting is not None:
+                hook_show_widget.set_text(hook_show_setting)
+        return
+
+    def _load_default_shell_settings(self):
         combo = self.get_widget('default_shell')
         # get the value for defualt shell. If unset, set to USER_SHELL_VALUE.
         value = self.client.get_string(KEY('/general/default_shell')) or USER_SHELL_VALUE
         for i in combo.get_model():
             if i[0] == value:
                 combo.set_active_iter(i.iter)
 
+    def _load_screen_settings(self):
+        """Load screen settings"""
+        # display number / use primary display
+        combo = self.get_widget('display_n')
+        dest_screen = self.client.get_int(KEY('/general/display_n'))
+        # If Guake is configured to use a screen that is not currently attached,
+        # default to 'primary display' option.
+        screen = self.get_widget('config-window').get_screen()
+        n_screens = screen.get_n_monitors()
+        if dest_screen > n_screens - 1:
+            self.client.set_bool(KEY('/general/mouse_display'), False)
+            dest_screen = screen.get_primary_monitor()
+            self.client.set_int(KEY('/general/display_n'), dest_screen)
+
+        if dest_screen == ALWAYS_ON_PRIMARY:
+            first_item = combo.get_model().get_iter_first()
+            combo.set_active_iter(first_item)
+        else:
+            seen_first = False  # first item "always on primary" is special
+            for i in combo.get_model():
+                if seen_first:
+                    i_int = int(i[0].split()[0])  # extracts 1 from '1' or from '1 (primary)'
```

Comment:
That looks sensible to potential translation, isn't there another way to retried this value ? (index?)